crates/subspace-farmer/src/ws_rpc.rs

```diff
@@ -46,25 +47,50 @@ impl RpcClient for WsRpc {
     }
 
     /// Subscribe to chain head.
-    async fn subscribe_new_head(&self) -> Result<Subscription<NewHead>, Error> {
-        self.client
+    async fn subscribe_new_head(&self) -> Result<mpsc::Receiver<NewHead>, Error> {
+        let mut subscription = self
+            .client
             .subscribe(
                 "chain_subscribeNewHead",
                 JsonRpcParams::NoParams,
                 "chain_unsubscribeNewHead",
             )
-            .await
+            .await?;
+
+        // used the default value for buffer from jsonrpsee
+        let (sender, receiver) = mpsc::channel(1024);
```

Comment:
Let's have a smaller buffer size here, even one should work, I suspect there is a buffer on the library side already anyway and skipping of these messages will not break anything.